src/include/nodes/primnodes.h

```diff
@@ -122,6 +123,7 @@ typedef struct IntoClause
 	Node	   *viewQuery;		/* materialized view's SELECT query */
 	bool		skipData;		/* true for WITH NO DATA */
 	Node       *distributedBy;  /* GPDB: columns to distribubte the data on. */
+	GpPolicy   *intoPolicy;		/* GPDB: save to restore for CTAS prepared */
```

Comment:
Can we use GpPolicy from other structure? The `create table t as select i from generate_series(1, 10) i distributed by (i);` query uses some structure to save GpPolicy. Can we use the same structure?

Reply:
I couldn't find any other way to get to `intoPolicy` in the right place.

Reply:
The IntoClause structure contains properties of a table to create, but GpPolicy is a property of a query which outputs rows for the table. I think it is strange that the same data are placed in different parts of the CreateTableAsStmt structure (the query and into fields).

Reply:
This is the only way I've found to pass `intoPolicy` from the parser to the planner to execute a prepared statement.

Reply:
what about calling setQryDistributionPolicy() in RevalidateCachedQuery()?

Reply:
1) `setQryDistributionPolicy` was already called in `transformCreateTableAsStmt`
2) it is a private function
3) it has an argument `ParseState *pstate`, which is not present in `RevalidateCachedQuery`

Reply:
reworked